.github/workflows/python-tests.yml

```diff
@@ -0,0 +1,72 @@
+name: Python Tests
+
+# 1. Run on every push 
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    # 2. Use the latest Ubuntu runner
+    runs-on: ubuntu-latest
+
+    # 3. Spin up a PostgreSQL service for tests
+    services:
+      postgres:
+        image: postgres:15
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: test_db
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd "pg_isready -U postgres"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      # 4. Check out code
+      - uses: actions/checkout@v3
+
+      # 5. Wait for Postgres to be ready
+      - name: Wait for Postgres
+        run: |
+          until pg_isready -h localhost -p 5432 -U postgres; do
+            echo "Waiting for Postgres…"
+            sleep 1
+          done
+
+      # 6. Create test database & apply schema
+      - name: Initialize database
+        env:
+          PGPASSWORD: postgres
+        run: |
+          psql -h localhost -U postgres -c "DROP DATABASE IF EXISTS test_db;"
+          psql -h localhost -U postgres -c "CREATE DATABASE test_db;"
+          psql -h localhost -U postgres -d test_db -f sql/01_create-tables.sql
+          psql -h localhost -U postgres -d test_db -f sql/03_create-view.sql
+
+      # 7. Set up Python
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
```

Comment:
given this is a new project, any reason to stick with 3.9?  i know we have 3.11 running some places and 3.13 is now published as stable.

@ivan-c do you have an opinion on best python version to run here (just testing, but ya know)?

Reply:
I usually use the version of python in debian stable as the lower bound, since debian stable is what we (should) have on most our own infra.

[that's currently 3.11](https://packages.debian.org/stable/python3), so we should start using that as the default at the moment (until the next debian release)